src/main/java/org/datadog/jenkins/plugins/datadog/DatadogUtilities.java

```diff
@@ -743,19 +711,29 @@ public static boolean isMainNode(String nodeName) {
     }
 
     public static String getNodeHostname(@Nullable EnvVars envVars, @Nullable Computer computer) {
-        if (envVars != null) {
-            String ddHostname = envVars.get(DatadogGlobalConfiguration.DD_CI_HOSTNAME);
-            if (DatadogUtilities.isValidHostname(ddHostname)) {
-                return ddHostname;
-            }
-            String hostname = envVars.get("HOSTNAME");
-            if (DatadogUtilities.isValidHostname(hostname)) {
-                return hostname;
+        if (computer != null) {
+            try {
+                EnvVars computerEnvironment = computer.getEnvironment();
+                String computerEnvVarsHostname = getNodeHostname(computerEnvironment);
+                if (isValidHostname(computerEnvVarsHostname)) {
+                    return computerEnvVarsHostname;
+                }
+
+            } catch (InterruptedException e) {
+                Thread.currentThread().interrupt();
+                logException(logger, Level.FINE, "Interrupted while trying to get computer env vars", e);
+            } catch (Exception e) {
+                logException(logger, Level.FINE, "Error getting computer env vars", e);
             }
         }
 
+        String envVarsHostname = getNodeHostname(envVars);
+        if (isValidHostname(envVarsHostname)) {
+            return envVarsHostname;
+        }
+
```

Comment:
It seems we're duplicating this piece of logic with computer EnvVars and other EnvVars.

Have we considered the following approach to avoid this duplicated code?
```
EnvVars finalEnvVars = computer != null ? compute.getEnvironment() : envVars
...
String envVarsHostname = getNodeHostname(envVars);
if (isValidHostname(envVarsHostname)) {
     return envVarsHostname;
}

```

Reply:
I would like to examine both, as I couldn't find a clear definition of what the "job-level" vars are: I suspect they might include whatever's being set in the Job config UI or in the pipeline DSL. The computer-level ones, however, include variables set externally on the worker node that executes the job.